src/Bicep.LangServer.IntegrationTests/CompletionTests.cs

```diff
@@ -1133,6 +1132,79 @@ static void AssertEqualsOperatorCompletion(CompletionItem item)
                 '|');
         }
 
+        [TestMethod]
+        public async Task TypeDrivenCompletionsAreOfferedInParameterAndOutputValues()
+        {
+            var fileWithCursors = @"
+type bigObject = {
+  foo: {
+    bar: {
+      baz: bool
+    }
+  }
+  fizz: {
+    buzz: {
+      pop: 'snap' | 'crackle'
+    }
+  }
+}
+
+param p bigObject = {
+  ǂ
+}
+
+param p2 bigObject = {
+  foo: {
+    ǂ
+  }
+}
+
+param p3 bigObject = {
+  foo: {
+    bar: {
+      ǂ
+    }
+  }
+}
+
+param p4 bigObject = {
+  foo: {
+    bar: {
+      baz: ǂ
+    }
+  }
+}
+
+output o bigObject = {
+  fizz: {
+    buzz: {
+      pop: ǂ
+    }
+  }
+}
+";
```

Comment:
Btw, we can replace verbatim string with [raw string literals](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/tokens/raw-string) since we are using C# 11 now.